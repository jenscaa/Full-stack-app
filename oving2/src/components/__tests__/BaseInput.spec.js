import { describe, it, expect } from 'vitest'
import { mount } from '@vue/test-utils'
import BaseInputVue from '../BaseInput.vue'

describe('BaseInput', () => {
    const wrapper = mount(BaseInputVue)

    it('renders properly', () => {
        expect(wrapper.html()).toMatchSnapshot()
    })

    it ('display label name', () => {
        const wrapper = mount(BaseInputVue, {
            props: {
                label: 'Name',
                error: ''
            }
        })
        expect(wrapper.find('label').text()).toBe('Name')
    })

    it ('display no error message', () => {
        const wrapper = mount(BaseInputVue, {
            props: {
                label: 'Name',
                error: ''
            }
        })
        expect(wrapper.find('p').exists()).toBe(false)
    })

    it('display error', () => {
        const wrapper = mount(BaseInputVue, {
            props: {
                label: 'Name',
                error: 'Error'
            }
        })
        expect(wrapper.find('p').text()).toBe('Error')
    })

    it('display textarea', () => {
        const wrapper = mount(BaseInputVue, {
            props: {
                label: 'Melding',
                error: 'Error'
            }
        })
        expect(wrapper.find('textarea').exists()).toBe(true)
    })

})