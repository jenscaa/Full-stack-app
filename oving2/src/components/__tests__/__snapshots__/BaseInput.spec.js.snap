// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BaseInput > renders properly 1`] = `
"<div data-v-82c69faf="" id="container">
  <div data-v-82c69faf="">
    <!--v-if-->
  </div>
  <div data-v-82c69faf=""><input data-v-82c69faf="" placeholder="" class="field"></div>
  <div data-v-82c69faf="">
    <!--v-if-->
  </div>
</div>"
`;
